node/routes.py

```diff
@@ -23,45 +23,42 @@ def get_db() -> dict[str, Any]:
     apps_data: dict[str, Any] = {}
 
     for app_name in list(zconfig.APPS.keys()):
-        sequenced_num: int = zdb.get_last_tx(app_name, "sequenced").get("index", 0)
-        locked_num: int = zdb.get_last_tx(app_name, "locked").get("index", 0)
-        finalized_num: int = zdb.get_last_tx(app_name, "finalized").get("index", 0)
-        all_num: int = len(zdb.apps[app_name]["transactions"])
+        sequenced_num: int = zdb.get_last_batch(app_name, "sequenced").get("index", 0)
+        locked_num: int = zdb.get_last_batch(app_name, "locked").get("index", 0)
+        finalized_num: int = zdb.get_last_batch(app_name, "finalized").get("index", 0)
+        all_num: int = len(zdb.apps[app_name]["batches"])
         apps_data[app_name] = {
-            "transactions_state": {
+            "batches_state": {
                 "sequenced": sequenced_num,
                 "locked": locked_num,
                 "finalized": finalized_num,
                 "all": all_num,
             },
             "nodes_state": zdb.apps[app_name]["nodes_state"],
-            "transactions": sorted(
-                list(zdb.apps[app_name]["transactions"].values()),
-                key=lambda tx: tx.get("index", 0),
+            "batches": sorted(
+                list(zdb.apps[app_name]["batches"].values()),
+                key=lambda batch: batch.get("index", 0),
             ),
         }
-
     return apps_data
 
 
 @node_blueprint.route("/transactions", methods=["PUT"])
 @utils.not_sequencer
 def put_transactions() -> Response:
-    """Put new transactions into the database."""
-    # TODO: only authorized users should be able to call this route
-    req_data: dict[str, Any] = request.get_json(silent=True) or {}
-    required_keys: list[str] = ["app_name", "transactions", "timestamp"]
-    error_message: str = utils.validate_request(req_data, required_keys)
+    """Put a new batch into the database."""
+    required_keys: list[str] = ["app_name"]
+    error_message: str = utils.validate_request(request.args, required_keys)
     if error_message:
         return error_response(ErrorCodes.INVALID_REQUEST, error_message)
-    zdb.init_txs(req_data["app_name"], req_data["transactions"])
+    zdb.init_batches(request.args.get('app_name'), [request.data.decode('utf-8')])
```

Comment:
This method can be changed to `init_batch` and get single batch
There is no need to pass the list anymore